src/desktop/src/index.js

```diff
@@ -78,35 +64,37 @@ initialiseStorage(getEncryptionKey)
         // Assign accountIndex to every account in accountInfo if it is not assigned already
         store.dispatch(assignAccountIndexIfNecessary(get(data, 'accounts.accountInfo')));
 
-        if (Electron.mode === 'tray') {
-            // Add Realm change listener to sync read-only Tray app with Main app
-            realm.addListener('change', () => {
-                const data = mapStorageToState();
-                store.dispatch(mapStorageToStateAction(data));
-            });
-        } else {
-            // Start Tray application if enabled in settings
-            const isTrayEnabled = get(data, 'settings.isTrayEnabled');
-            Electron.setTray(isTrayEnabled);
-        }
+        // Proxy realm changes to Tray application
+        realm.addListener('change', () => {
+            const data = mapStorageToState();
+            Electron.storeUpdate(JSON.stringify(data));
+        });
+
+        // Start Tray application if enabled in settings
+        const isTrayEnabled = get(data, 'settings.isTrayEnabled');
+        Electron.setTray(isTrayEnabled);
 
-        render(
-            <ErrorBoundary>
-                <Redux store={store}>
-                    <I18nextProvider i18n={i18next}>
-                        <Router>
-                            {Electron.mode === 'tray' ? (
-                                <Tray />
-                            ) : (
-                                <React.Fragment>
-                                    <Alerts />
-                                    <Index />
-                                </React.Fragment>
-                            )}
-                        </Router>
-                    </I18nextProvider>
-                </Redux>
-            </ErrorBoundary>,
-            document.getElementById('root'),
-        );
+        // Show Wallet window after inital store update
+        Electron.focus();
     });
```

Comment:
Since the render method is now out of the promise chain, I think the errors should be caught here. In development mode, perhaps we can just log the error.